### ISSUE TITLE
rbindlist silently dropping NA factor

### ISSUE THREAD

**Issue by DavorJ** [Rdatatable/data.table#4425]
```
> df1 <- data.frame(col = factor(c('A', NA, 'B'), exclude = NULL))
> df1
   col
1    A
2 <NA>
3    B

> df2 <- data.frame(col = factor('C'))

> df <- data.table::rbindlist(list(df1, df2))
> levels(df$col)
[1] "A" "B" "C"
```

The documentation for `data.table::rbindlist()` reads:

> If a column contains factors then a factor is created. [...] Any strings in regular factor and character columns which are missing from the longest ordered levels are added at the end.

Unfortunately, `NA` factors are silently dropped. Is this by design and intended? Because I don't expect any factors to be dropped.

I am using `data.table_1.12.4`.

**Comment by jangorecki** [comment]
Actually `NA` is not a string, even `NA_character_` is not a string. It is a marker that value is not available. But agree that could be handled similarly are ordinary string factor levels.

**Comment by jangorecki** [comment]
Worth to note we are now consistent to base R do.call-rbind, so when addressing this request it would be good to mention that inconsistency in manual.
```r
df = do.call("rbind", list(df1, df2))
levels(df$col)
#[1] "A" "B" "C"
```

**Comment by DavorJ** [comment]
> for now I suggest to use the following post-processing after rbindlist
> 
> ```r
> df[, col := addNA(col)]
> ```

I agree with one remark though: I expect that the factor levels of the first dataframe will be unchanged, and the rest added (as the docs suggest). But since `NA` is excluded, this mixes up the levels. Adding it at the end doesn't solve the mix-up. 

I use this technique to rbind data where my first dataframe is a "skelet" containing the correct numerical encodings of all factors. There might be better techniques for this, but they are unknown to me.

**Comment by ben-schwen** [comment]
This is by design upon request, see also #3479. Adding an additional argument seems possible though e.g. `nalevels = TRUE\FALSE`
